distributed/worker.py

```diff
@@ -2229,13 +2231,21 @@ def trigger_profile(self):
             active_threads = self.active_threads.copy()
         frames = sys._current_frames()
         frames = {ident: frames[ident] for ident in active_threads}
+        llframes = {}
+        if self.low_level_profiler:
+            llframes = {ident: profile.ll_get_stack(ident) for ident in active_threads}
         for ident, frame in frames.items():
             if frame is not None:
                 key = key_split(active_threads[ident])
-                profile.process(frame, None, self.profile_recent,
-                                stop='distributed/worker.py')
-                profile.process(frame, None, self.profile_keys[key],
+                init = profile.create()
+                llframe = llframes.get(ident)
+
+                state = profile.process(frame, True, self.profile_recent,
                                 stop='distributed/worker.py')
+                state = profile.llprocess(llframe, None, state)
```

Comment:
We might want to defend the `llframe` lines with a `if self.low_level_profiler` conditional.

Reply:
Yes, I'll do that, thanks for pointing it out!

Reply:
After looking at the code, I remember why I didn't do that. `llframes` is an empty dict when `self.low_level_profiler = False`, therefore, calling `llframes.get(ident)` returns `None`, and `profile.llprocess()` starts with `if not frames: return`. We could still add the conditional here, but it would be redundant. I would normally not do that in code I write, but if you think it's useful for someone reading the code, it's fine by me.